airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/FileSystemConfigPersistence.java

```diff
@@ -57,14 +57,13 @@
   // root for where configs are stored
   private final Path configRoot;
 
-  public static ConfigPersistence createWithValidation(final Path storageRoot) throws IOException {
+  public static ConfigPersistence createWithValidation(final Path storageRoot) {
     return new ValidatingConfigPersistence(new FileSystemConfigPersistence(storageRoot));
   }
 
-  public FileSystemConfigPersistence(final Path storageRoot) throws IOException {
+  public FileSystemConfigPersistence(final Path storageRoot) {
     this.storageRoot = storageRoot;
     this.configRoot = storageRoot.resolve(CONFIG_DIR);
-    Files.createDirectories(configRoot);
```

Comment:
This was added by me in #4670. But I should not have done that. This statement covered the corner case that sometimes the config volume is not ready when server app starts, at least in kube acceptance test.